src/components/Executions/utils.ts

```diff
@@ -114,96 +115,81 @@ export const taskExecutionIsTerminal = (taskExecution: TaskExecution) =>
     taskExecution.closure &&
     terminalTaskExecutionStates.includes(taskExecution.closure.phase);
 
-/** Assigns display information to NodeExecutions. Each NodeExecution has an
- * associated `nodeId`. If a `Workflow` is provided, node/task information will
- * be pulled from the workflow closure to determine the node type.
- */
-export function mapNodeExecutionDetails(
-    executions: NodeExecution[],
-    workflow?: Workflow
+/** Populates a NodeExecution with extended information read from an `ExecutionDataCache` */
+export function populateNodeExecutionDetails(
+    nodeExecution: NodeExecution,
+    dataCache: ExecutionDataCache
 ) {
-    let nodesById: Dictionary<CompiledNode> = {};
-    let taskTemplates: Dictionary<TaskTemplate> = {};
+    const { nodeId } = nodeExecution.id;
+    const cacheKey = getCacheKey(nodeExecution.id);
+    const nodeInfo = dataCache.getNodeForNodeExecution(nodeExecution.id);
 
-    if (workflow) {
-        if (!workflow.closure) {
-            throw new Error('Workflow has no closure');
+    let displayId = nodeId;
+    let displayType = NodeExecutionDisplayType.Unknown;
+    let taskTemplate: TaskTemplate | undefined = undefined;
+
+    if (nodeInfo == null) {
```

Comment:
Is the double-equal syntax here expected? Or can we utilize `===`

Reply:
`==` is checking for null _or_ undefined. I agree that it is not exactly obvious.